src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -0,0 +1,88 @@
+import { CurrencyAmount, Currency, Price, Token } from '@uniswap/sdk-core'
+import { useEffect, useMemo, useState } from 'react'
+import { unstable_batchedUpdates as batchedUpdate } from 'react-dom'
+import { supportedChainId } from 'utils/supportedChainId'
+import { useActiveWeb3React } from 'hooks/web3'
+import { getBestPrice } from 'utils/price'
+import { STABLECOIN_AMOUNT_OUT } from 'hooks/useUSDCPrice'
+import { stringToCurrency } from 'state/swap/extension'
+
+export * from '@src/hooks/useUSDCPrice'
+export { default } from '@src/hooks/useUSDCPrice'
+
+export function useBestUSDCPrice(currency?: Currency) {
+  const [bestUsdPrice, setBestUsdPrice] = useState<Price<Token, Currency> | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+
+  const { chainId } = useActiveWeb3React()
+
+  // USDC constants
+  // 100_000e6 USDC @ 6 decimals
+  const amountOut = chainId ? STABLECOIN_AMOUNT_OUT[chainId] : undefined
+  const stablecoin = amountOut?.currency
+
+  useEffect(() => {
+    const isSupportedChain = supportedChainId(chainId)
+    if (!isSupportedChain || !currency || !amountOut || !stablecoin) return
+
+    const params = {
+      baseToken: stablecoin.address,
+      quoteToken: currency.wrapped.address,
+      kind: 'buy',
+      amount: amountOut.quotient.toString(),
+      chainId: isSupportedChain,
+      fromDecimals: currency.decimals,
+      toDecimals: stablecoin.decimals,
+    }
+
+    getBestPrice(params, { aggrOverride: 'max' })
+      .then((winningPrice) => {
+        // Response can include a null price amount, throw if so
+        if (!winningPrice.amount) throw new Error('Winning price cannot be null')
```

Comment:
why would be null? afaik, it would fail if it doesn't find a price.

Reply:
Also, what happen with this error, who is handling it?

Reply:
price can be `null` if the `fee > inputAmt` e.g i think? that's the type returned by getBestPrice so I am just handling it (badly)

Reply:
The error is mainly just used as a check later in the next hook, logged to console for now. Not really sure how best to handle an error. maybe just return null USD price?

Reply:
f51d601

return null price for USDC